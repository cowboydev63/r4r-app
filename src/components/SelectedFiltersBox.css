.selected-filters__header.r4r-DEFAULT {
    min-width: 120px;
    font-size: 16px;
    align-self: flex-start;
    margin: 6px 0;
}

.selected-filters__filters-container.r4r-DEFAULT {
    display: flex;
    flex-flow: row wrap;
    align-items: center;
}

.selected-filters__filter.r4r-DEFAULT {
    color: white;
    background: #2a71a5;
    font-size: 12px;
    padding: 1px 7px;
    margin: 3px 5px;
    border-radius: 6px;
    cursor: pointer;
    display: flex;
    justify-content: space-around;
}

.selected-filters__filter.selected-filters__clear.r4r-DEFAULT {
    background: none;
    color: #169bd5;
    font-size: 16px;
    width: auto;
    margin-top: 0;
    margin-bottom: 0;
}

.selected-filters__filter.r4r-DEFAULT > p:nth-child(2) {
    width: 10px;
    margin-left: 10px;
    text-align: center;
    font-weight: bold;
}