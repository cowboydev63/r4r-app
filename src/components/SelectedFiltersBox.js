import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';
import {
    keyHandler,
} from '../utilities';
import {
    filterInterface
} from '../interfaces';
import './SelectedFiltersBox.css';

class SelectedFiltersBox extends React.PureComponent {
    static propTypes = {
        selected: PropTypes.arrayOf(filterInterface).isRequired, 
        toggleFilter: PropTypes.func.isRequired,
        clearFilters: PropTypes.func.isRequired,
    }

    static defaultProps = {
        selected: [],
    }

    render(){
        if(!this.props.selected.length){
            return null;
        }

        return (
            <Theme element="section" className="results__selected-filters" aria-label="Selected Search Filters">
                {
                    this.props.selected.length
                    ?
                        <React.Fragment>
                            <Theme element="p" className="selected-filters__header" aria-hidden>Your Selections:</Theme>
                            <Theme element="div" className="selected-filters__filters-container">
                            {
                                this.props.selected.map((filter, idx) => (
                                    <Theme
                                        element="div" 
                                        key={ idx }
                                        className="selected-filters__filter"
                                        tabIndex="0"
                                        onClick={ this.props.toggleFilter(filter.param)(filter.key) }
                                        onKeyPress={ keyHandler({
                                            fn: this.props.toggleFilter(filter.param)(filter.key),
                                        })}
                                    >
                                        <p>{filter.label}</p>
                                        <p>âœ–</p>
                                    </Theme>
                                ))
                            }
                            {
                                (this.props.selected.length > 1) &&
                                    <Theme
                                        element="div"
                                        tabIndex="0"
                                        className="selected-filters__filter selected-filters__clear"
                                        onClick={ this.props.clearFilters }
                                        onKeyPress={ keyHandler({
                                            fn: this.props.clearFilters,
                                        })}
                                    >
                                        <p>Clear all</p>
                                    </Theme>
                            }
                            </Theme>
                        </React.Fragment>
                    :
                        null
                }
            </Theme>
        )

    }
}

export default SelectedFiltersBox;