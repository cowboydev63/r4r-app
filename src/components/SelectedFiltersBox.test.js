import React from 'react';
import { shallow } from 'enzyme';
import SelectedFiltersBox from './SelectedFiltersBox';

describe('SelectedFiltersBox Component', () => {
  it('renders without crashing', () => {
    shallow(<SelectedFiltersBox toggleFilter={()=> {}} clearFilters={() => {}}/>);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<SelectedFiltersBox toggleFilter={()=> {}} clearFilters={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  })
})