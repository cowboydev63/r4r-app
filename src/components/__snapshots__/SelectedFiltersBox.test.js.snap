// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFiltersBox Component renders correctly 1`] = `""`;
